src/swupd_lib/target_root.c

```diff
@@ -330,6 +333,61 @@ static enum swupd_code install_dir(const char *fullfile_path, const char *target
 	return install_dir_using_tar(fullfile_path, target_file);
 }
 
+/* Backup target directory to a backup and timestamp prefixed name. */
+/* This function should only be used after the deletes are processed. */
+static enum swupd_code move_to_timestamp(const char *filename)
```

Comment:
maybe call it "dirname" not "filename" to indicate we expect it to be a directory?